app/src/main/java/fr/free/nrw/commons/location/LocationPermissionsHelper.java

```diff
@@ -15,7 +15,26 @@
 import fr.free.nrw.commons.utils.PermissionUtils;
 
 /**
- * Helper class to handle location permissions
+ * Helper class to handle location permissions.
+ *
+ * Location flow for fragments containing a map is as follows:
+ * Case 1: When location permission has never been asked for or denied before
+ * Check if permission is already granted or not.
+ * If not already granted, ask for it (if it isn't denied twice before).
+ * If now user grants permission, go to Case 3/4, else go to Case 2.
+ *
+ * Case 2: When location permission is just asked but has been denied
+ * Shows a toast to tell the user why location permission is needed.
+ * Show current location / nearby pins / nearby images according to the default location.
```

Comment:
I think we actually go back to step 1 after the user agrees to the rationale. Could you mention the same?

Reply:
Done